reframe/utility/typecheck.py

```diff
@@ -129,15 +140,20 @@ def __getitem__(cls, elem_type):
             raise TypeError('invalid type specification for container type: '
                             'expected ContainerType[elem_type]')
 
-        ret = _ContainerType('%s[%s]' % (cls.__name__, elem_type.__name__),
-                             cls._bases, cls._namespace)
+        ret = _SequenceType('%s[%s]' % (cls.__name__, elem_type.__name__),
+                            cls._bases, cls._namespace)
         ret._elem_type = elem_type
-        ret.register_subtypes()
+        ret.register_container_type()
         cls.register(ret)
         return ret
 
+    def __rfm_cast__(cls, s):
+        container_type = cls._type
+        elem_type = cls._elem_type
+        return container_type(elem_type(e) for e in s.split(','))
```

Comment:
I think the special treatment for certain arguments passed to the call method can go here. This would allow to extend the supported conversions on each container independently. If the `*args` & `**kwargs` do not match any implemented conversion, then just return `cls(*args, **kwargs)`.

Reply:
I don't understand very well this. Is it something more general related to the comment above or specific to this call?

Reply:
Yeah, this was related to the comment above, so all good.